doc/classes/Object.xml

```diff
@@ -1,33 +1,38 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="Object" version="4.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
 	<brief_description>
-		Base class for all non-built-in types.
+		Base class for all objects in the engine.
 	</brief_description>
 	<description>
-		Every class which is not a built-in type inherits from this class.
-		You can construct Objects from scripting languages, using [code]Object.new()[/code] in GDScript, or [code]new Object[/code] in C#.
-		Objects do not manage memory. If a class inherits from Object, you will have to delete instances of it manually. To do so, call the [method free] method from your script or delete the instance from C++.
-		Some classes that extend Object add memory management. This is the case of [RefCounted], which counts references and deletes itself automatically when no longer referenced. [Node], another fundamental type, deletes all its children when freed from memory.
-		Objects export properties, which are mainly useful for storage and editing, but not really so much in programming. Properties are exported in [method _get_property_list] and handled in [method _get] and [method _set]. However, scripting languages and C++ have simpler means to export them.
-		Property membership can be tested directly in GDScript using [code]in[/code]:
+		An advanced [Variant] type. All classes in the engine inherit from Object. Each class may define new properties, methods or signals, which are available to all inheriting classes. For example, a [Sprite2D] instance is able to call [method Node.add_child] because it inherits from [Node].
+		You can create new instances, using [code]Object.new()[/code] in GDScript, or [code]new Object[/code] in C#.
+		To delete an Object instance, call [method free]. This prevents memory leaks, and is necessary for most classes inheriting Object, because they do not manage memory on their own. [RefCounted] is the biggest exception to this. [Node], another essential object, also deletes its children when freed.
```

Comment:
> To delete an Object instance, call [method free]. This prevents memory leaks...

Nitpick: the memory leak part is a bit strange. It sounds like:

```gdscript
if you_want_delete_an_object():
    please_call_free()
    assert(not have_memory_leaks())
else:
    pass  # Most cases of memory leaks happen here, i.e. you forget to delete it.
```

This is technically correct, but memory leaks are caused by not deleting objects, not a result of not calling `free()`.

```suggestion
		To delete an Object instance manually, call [method free] or [method queue_free]. This is necessary for most classes inheriting Object, because they do not manage memory on their own. There are a few classes that perform memory management. For example, [RefCounted] deletes itself when no longer referenced and [Node] deletes its children when freed. Please make sure an object is deleted when it is no longer needed, as this will prevent memory leaks.
```

Reply:
I know it may sound incredibly strange but Object by itself does not have `queue_free`.

Reply:
Ah, yeah. I forget it's a Node method.